src/spec/junit_formatter.cr

```diff
@@ -100,7 +100,11 @@ module Spec
     end
 
     private def classname(result)
-      result.file.sub(%r{\.[^/.]+\Z}, "").gsub("/", ".").gsub(/\A\.+|\.+\Z/, "")
+      path = result.file
+      if relative_path = path.lchop?(Dir.current).try(&.lchop('/'))
```

Comment:
This path handling is very crude and will fail in a few edge cases.

Both paths need at least to be normalized before comparison. 
Unfortunately, we're still missing an efficient method to check if one path is a parent of another. `Path.new(path).parents.includes?(Path.new(Dir.current).normalize)` should to for now.

Reply:
> This path handling is very crude and will fail in a few edge cases.

Could you specify those edge cases?

> Both paths need at least to be normalized before comparison.

There's no need to, since both are absolute normalized paths.

> Unfortunately, we're still missing an efficient method to check if one path is a parent of another. `Path.new(path).parents.includes?(Path.new(Dir.current).normalize)` should to for now.

True, but imho that's different issue.

Reply:
> Could you specify those edge cases?

Mostly windows paths, such as `\` as path separator, UNC shares etc.

> There's no need to, since both are absolute normalized paths.

How do you know? `Result#file` can be an almost arbitrary path. And I'm not sure `Dir#current` makes any guarantees for being normalized.

Adding a proper method is a different issue, but here we should use at least a correct workaround.

Reply:
Yes, this implementation won't handle such edge-cases indeed - as it was the case with the old one, btw.

Reply:
The previous implementation didn't even handle relative path resolution. Nevertheless a new solution should be correct anyway.

Reply:
I've force-pushed sth which should work as u mentioned.